Week3/homework/step2-2.js

```diff
@@ -4,17 +4,30 @@ function threeFive(startIndex, stopIndex, threeCallback, fiveCallback) {
   const numbers = [];
 
   // Replace this comment and the next line with your code
+  for (let i = startIndex; i <= stopIndex; i++) {
+    numbers.push(i);
+    if (i % 3 === 0 || i % 5 === 0) {
+      if (i % 3 === 0 && i % 5 === 0) {
+        threeCallback(i);
+        fiveCallback(i);
+      } else if (i % 3 === 0) {
+        threeCallback(i);
+      } else {
+        fiveCallback(i);
+      }
```

Comment:
Ah. I think you can improve the logic here a bit more. For example, if you think mathematically, a number which is divisible by both 3 and 5 will always be divisible by number 15, isn't it? Also, you don't really need a nested-if block here. You could rewrite the logic as:

```
if (i % 15 === 0) {
  // call both three and five callbacks
} else if(i % 3 === 0) {
  // call three callbacks
} else if(i % 5 === 0) {
  // call five callbacks
}
```